Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -263,40 +262,40 @@ bool CFileSystemGCWii::DetectFileSystem()
 void CFileSystemGCWii::InitFileSystem()
 {
 	m_Initialized = true;
+	u32 const shift = GetOffsetShift();
 
 	// read the whole FST
-	u64 FSTOffset = (u64)Read32(0x424) << GetOffsetShift();
+	u64 FSTOffset = static_cast<u64>(Read32(0x424)) << shift;
 	// u32 FSTSize     = Read32(0x428);
 	// u32 FSTMaxSize  = Read32(0x42C);
 
 
 	// read all fileinfos
-	SFileInfo Root;
-	Root.m_NameOffset = Read32(FSTOffset + 0x0);
-	Root.m_Offset     = (u64)Read32(FSTOffset + 0x4) << GetOffsetShift();
-	Root.m_FileSize   = Read32(FSTOffset + 0x8);
-
-	if (Root.IsDirectory())
+	SFileInfo Root
 	{
-		if (m_FileInfoVector.size())
-			PanicAlert("Wtf?");
-		u64 NameTableOffset = FSTOffset;
+		Read32(FSTOffset + 0x0),
+		static_cast<u64>(FSTOffset + 0x4) << shift,
+		Read32(FSTOffset + 0x8)
+	};
 
-		m_FileInfoVector.reserve((unsigned int)Root.m_FileSize);
-		for (u32 i = 0; i < Root.m_FileSize; i++)
-		{
-			SFileInfo sfi;
-			u64 Offset = FSTOffset + (i * 0xC);
-			sfi.m_NameOffset = Read32(Offset + 0x0);
-			sfi.m_Offset     = (u64)Read32(Offset + 0x4) << GetOffsetShift();
-			sfi.m_FileSize   = Read32(Offset + 0x8);
-
-			m_FileInfoVector.push_back(sfi);
-			NameTableOffset += 0xC;
-		}
+	if (!Root.IsDirectory()) { return; }
```

Comment:
Bodies go on the next line, so this would be:

```cpp
if (!Root.IsDirectory())
{
    return;
}
```

or 
```cpp
if (!Root.IsDirectory())
    return;
```